### PR TITLE
DBFTPlugin: adapt Conflicts attribute verification

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Depends on https://github.com/neo-project/neo/pull/2818, see the https://github.com/neo-project/neo/pull/2818#issuecomment-1548814976.

@vncoelho, we'll need to add the proper link to the core module to .csproj file to build this PR. Could you test it, please?

### CODE REVIEW

**Comment by vncoelho** [comment]
Yes, I will build and test. 
I will create a script for serializing txs. Did you try it as well?

**Comment by AnnaShaleva** [comment]
> Did you try it as well?

Not yet, only unit-tests. I'll try to run four-nodes private network with two C# and two neo-go nodes with the help of neo-bench and see how it goes.

File: src/DBFTPlugin/Consensus/ConsensusService.cs [hunk 1]
```diff
@@ -257,7 +257,38 @@ private bool AddTransaction(Transaction tx, bool verify)
...
+                // need to check that current context doesn't contain conflicting transactions.
+                VerifyResult result;
+
+                // Firstly, check whether tx has Conlicts attribute with the hash of one of the context's transactions.
+                foreach (var h in tx.GetAttributes<Conflicts>().Select(attr => attr.Hash))
+                {
+                    if (context.TransactionHashes.Contains(h))
+                    {
+                        result = VerifyResult.HasConflicts;
+                        Log($"Rejected tx: {tx.Hash}, {result}{Environment.NewLine}{tx.ToArray().ToHexString()}", LogLevel.Warning);
```
Comment (shargon) [inline src/DBFTPlugin/Consensus/ConsensusService.cs:271]:
```suggestion
                        Log($"Rejected conflict tx: {tx.Hash}, {result}{Environment.NewLine}{tx.ToArray().ToHexString()}", LogLevel.Warning);
```

File: src/DBFTPlugin/Consensus/ConsensusService.cs [hunk 2]
```diff
@@ -257,7 +257,38 @@ private bool AddTransaction(Transaction tx, bool verify)
...
+                        return false;
+                    }
+                }
+                // After that, check whether context's transactions have Conflicts attribute with tx's hash.
+                foreach (var pooledTx in context.Transactions.Values)
+                {
+                    if (pooledTx.GetAttributes<Conflicts>().Select(attr => attr.Hash).Contains(tx.Hash))
+                    {
+                        result = VerifyResult.HasConflicts;
+                        Log($"Rejected tx: {tx.Hash}, {result}{Environment.NewLine}{tx.ToArray().ToHexString()}", LogLevel.Warning);
```
Comment (shargon) [inline src/DBFTPlugin/Consensus/ConsensusService.cs:282]:
```suggestion
                        Log($"Rejected conflict tx: {tx.Hash}, {result}{Environment.NewLine}{tx.ToArray().ToHexString()}", LogLevel.Warning);
```

**Review by shargon — COMMENTED** [review]
Wait for https://github.com/neo-project/neo/pull/2818

File: src/DBFTPlugin/Consensus/ConsensusService.cs [hunk 3]
```diff
@@ -257,7 +257,38 @@ private bool AddTransaction(Transaction tx, bool verify)
         {
             if (verify)
             {
-                VerifyResult result = tx.Verify(neoSystem.Settings, context.Snapshot, context.VerificationContext);
+                // At this step we're sure that there's no on-chain transaction that conflicts with
```
Comment (vncoelho) [inline src/DBFTPlugin/Consensus/ConsensusService.cs:260]: we're sure -> we ensure

**Comment by AnnaShaleva** [comment]
@superboyiii, I've updated the dBFT plugin based on your test and https://github.com/neo-project/neo/pull/2818#issuecomment-1562554901. You'll need to use the updated version in your test.

**Comment by roman-khimov** [comment]
@superboyiii, @shargon, just a reminder, but this needs to merged as well before 3.6 release. Otherwise neo-project/neo#2818 won't be completely functional.

**Pull request merged**